vyper/semantics/analysis/__init__.py

```diff
@@ -12,6 +13,7 @@ def validate_semantics(vyper_ast, input_bundle):
     namespace = get_namespace()
 
     with namespace.enter_scope():
+        pre_typecheck(vyper_ast)
         add_module_namespace(vyper_ast, input_bundle)
-        vy_ast.expansion.expand_annotated_ast(vyper_ast)
+        vy_ast.expansion.generate_public_variable_getters(vyper_ast)
```

Comment:
Getter functions are added before semantics validation so they can be annotated.